core/src/database/DatabaseSessionPool.cpp

```diff
@@ -39,7 +39,8 @@ namespace ledger {
             const std::shared_ptr<api::PathResolver> &resolver,
             const std::shared_ptr<spdlog::logger>& logger,
             const std::string &dbName,
-            const std::string &password
+            const std::string &password,
+            bool usingPostgreSQLDatabase
```

Comment:
The backend is already provided as argument… that `bool` is suspicious to me in terms of API.